modules/apps/dynamic-data-mapping/dynamic-data-mapping-form-field-type/src/main/resources/META-INF/resources/date/date_field.js

```diff
@@ -3,6 +3,29 @@ AUI.add(
 	function(A) {
 		var isArray = Array.isArray;
 
+		var languageId = Liferay.ThemeDisplay.getLanguageId().replace('_', '-');
+		var customDateFormat = A.Intl.get('datatype-date-format', 'x', languageId);
+		var dateFormat = Liferay.AUI.getDateFormat();
```

Comment:
You could make this more efficient if you didn't initialize dateformat until after you do the check for customdateformat.

The change could be to something like:
var dateFormat;
if (customDateFormat) {
	dateFormat = customDateFormat;
}
else {
       dateFormat = Liferay.AUI.getDateFormat();
}

This proposed change would therefore make it so you only have to run getDateFormat if there's no customDateFormat.

Reply:
Thanks, will change this